### PR TITLE
Add impl `http_body::Body` for `hyper::Body`

### PR DESCRIPTION
**Pull request by LucioFranco**
This adds a `http_body::Body` impl for hypers `Body`. This should
allow us to start moving to a more generic body trait based on
`BufStream` and `http-body`.

cc @seanmonstar

### CODE REVIEW

**Comment by LucioFranco** [comment]
GitCop 😭 idk what is wrong.

**Comment by seanmonstar** [comment]
Don't worry about GitCop, I always fix it as part of the squash step.

**Comment by LucioFranco** [comment]
@seanmonstar so do you suggest a blanket impl of Body for T: Payload? Otherwise those changes seem fine, will get to them soon.

**Comment by seanmonstar** [comment]
No, I mean instead of having `poll_trailers` have duplicate code, do like this:

```rust
impl http_body::Body for Body {
    fn poll_trailers(&mut self) -> Poll<Option<HeaderMap>, Self::Error> {
        Payload::poll_trailers(self)
    }
    // etc...
}
```

**Comment by LucioFranco** [comment]
@seanmonstar I think I've addressed all your comments. Should be good for another pass.

File: src/body/body.rs [hunk 1]
```diff
@@ -332,6 +322,35 @@ impl Payload for Body {
...
+        <Self as Payload>::is_end_stream(self)
+    }
+
+    fn size_hint(&self) -> SizeHint {
+        let mut hint = SizeHint::default();
+
+        let upper_limit = <Self as Payload>::content_length(self);
+
+        if let Some(upper) = upper_limit {
+            hint.set_upper(upper);
```
Comment (seanmonstar) [inline src/body/body.rs]: I believe we can set the lower as well, since it's meant to be an exact size in `Payload`.

File: src/body/body.rs [hunk 2]
```diff
@@ -183,54 +183,44 @@ impl Body {
...
                 delayed_eof: None,
                 on_upgrade: OnUpgrade::none(),
-            }))
+            })
+        })
     }
 
     fn poll_eof(&mut self) -> Poll<Option<Chunk>, ::Error> {
         match self.take_delayed_eof() {
-            Some(DelayEof::NotEof(mut delay)) => {
```
Comment (seanmonstar) [inline src/body/body.rs]: Just checking, this is simply rustfmt?

Comment (LucioFranco) [reply src/body/body.rs]: Ah damn totally could have been forgot to disable it for this PR oops. Would you like me to revert this?

File: src/body/body.rs (hunk 1 above)
Comment (LucioFranco) [reply src/body/body.rs]: so if I understand correctly, since we ahve the total content_length here so the exact size, lower would == upper?

Comment (seanmonstar) [reply src/body/body.rs]: Yep, `Payload::content_length` is an exact size if `Some(_)`.

**Comment by LucioFranco** [comment]
@seanmonstar I addressed your comment and reverted the rustfmt. Only issue is that h2 introduced breaking changes for 1.26. We can try to pin it to an older version?

**Comment by seanmonstar** [comment]
Just finished publishing a new h2 version that should work on Rust 1.26, so I gave CI a nudge.

**Comment by seanmonstar** [comment]
Woot, green! Thanks!

**Pull request merged**